File: lib-cli/tmp/server/producer/responses/users_request.rs

use super::{identification, producer::Control, protocol, Context};
use fiber::server;

#[allow(unused_variables)]
pub async fn response<E: std::error::Error, C: server::Control<E> + Send + Clone>(
    identification: &mut identification::Identification,
    filter: &identification::Filter,
    context: &mut Context,
    request: &protocol::Users::Request,
    control: &Control<E, C>,
) -> Result<protocol::Users::Response, protocol::Users::Err> {
    panic!("Handler for protocol::Users::Request isn't implemented");
}

File: lib-cli/tmp/server-ts/src/producer/responses/users.request.ts
import { Response } from "../implementation/responses/users.request";
import {
	Context,
	Producer,
	Identification,
	Filter,
	Protocol,
} from "../implementation/responses";

export function response(
	request: Protocol.Users.Request,
	consumer: Identification,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<Response> {
	return Promise.resolve(
		new Response(
			new Protocol.Users.Response({
				users: context.getUsers(),
			})
		)
	);
	// return Promise.reject(
	// 	new Error(`Handler for Protocol.Users.Request isn't implemented.`)
	// );
}
